fe/src/main/java/org/apache/doris/load/DeleteHandler.java

```diff
@@ -0,0 +1,684 @@
+                    db.writeUnlock();
+                }
+                Thread.sleep(1000);
+            }
+        } catch (Exception e) {
+            String failMsg = "delete unknown, " + e.getMessage();
+            LOG.warn(failMsg, e);
+            throw new DdlException(failMsg);
+        }
+    }
+
+    public class DeleteTaskChecker extends Thread {
+        private BlockingQueue<DeleteTask> queue;
+
+        public DeleteTaskChecker(BlockingQueue<DeleteTask> queue) {
+            this.queue = queue;
+        }
+
+        @Override
+        public void run() {
+            LOG.info("delete task checker start");
+            try {
+                loop();
+            } finally {
+                synchronized(queue) {
+                    queue.clear();
+                }
+            }
+        }
+
+        public void loop() {
+            while (true) {
+                try {
+                    DeleteTask task = queue.take();
+                    while (!task.isQuorum()) {
+                        long signature = task.getSignature();
+                        if (task.isCancel()) {
+                            break;
+                        }
+                        if (!executor.submit(task)) {
+                            Thread.sleep(1000);
+                            continue;
+                        }
+                        // re add to the tail
+                        queue.put(task);
+                    }
+                    // remove task isQuorum or isCanceled
+                    removeTask(task);
+                } catch (InterruptedException e) {
+                    // do nothing
+                }
+            }
+        }
+    }
+
+    private void commitTask(DeleteTask task, Database db) {
+        long transactionId = task.getSignature();
+        GlobalTransactionMgr globalTransactionMgr = Catalog.getCurrentGlobalTransactionMgr();
+        TransactionState transactionState = globalTransactionMgr.getTransactionState(transactionId);
+        List<TabletCommitInfo> tabletCommitInfos = new ArrayList<TabletCommitInfo>();
+        db.writeLock();
+        try {
+            TabletInvertedIndex invertedIndex = Catalog.getCurrentInvertedIndex();
+            for (TabletDeleteInfo tDeleteInfo : task.getTabletDeleteInfo()) {
+                for (Replica replica : tDeleteInfo.getFinishedReplicas()) {
+                    // the inverted index contains rolling up replica
+                    Long tabletId = invertedIndex.getTabletIdByReplica(replica.getId());
+                    if (tabletId == null) {
+                        LOG.warn("could not find tablet id for replica {}, the tablet maybe dropped", replica);
+                        continue;
+                    }
+                    tabletCommitInfos.add(new TabletCommitInfo(tabletId, replica.getBackendId()));
+                }
+            }
+            globalTransactionMgr.commitTransaction(db.getId(), transactionId, tabletCommitInfos);
```

Comment:
use `commitAndPublishTransaction()` instead, and then `afterCommit()` can be removed.

Reply:
done